### PR TITLE
feat(ads): placement bidders

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Updates the **Advertising -> Placements** UI to support registered bidders placement IDs.

### How to test the changes in this Pull Request:

See https://github.com/Automattic/newspack-ads/pull/231

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by thomasguillot** [comment]
LGTM 👍

**Comment by miguelpeixe** [comment]
Thank you, @thomasguillot!

File: assets/wizards/advertising/views/placements/index.js [hunk 1]
```diff
@@ -47,16 +51,122 @@ const getAdUnitsForSelect = adUnits => {
...
 	const [ placements, setPlacements ] = useState( {} );
-	const placementsApiFetch = options => {
-		setInFlight( true );
-		apiFetch( options )
+	const [ bidders, setBidders ] = useState( {} );
+	const [ biddersError, setBiddersError ] = useState( null );
+
+	const placementsApiFetch = async ( options, quiet = false ) => {
+		if ( ! quiet ) setInFlight( true );
+		await apiFetch( options )
```
Comment (dkoo) [inline assets/wizards/advertising/views/placements/index.js:169]: This is purely a non-blocking, subjective code-style thing, but what's the purpose of using `async`/`await` if still using promises syntax? I say either use promises or `async`/`await` with synchronous-style syntax, e.g.:

```js
const placementsApiFetch = async ( options, quiet = false ) => {
	if ( ! quiet ) setInFlight( true );
	try {
		const data = await apiFetch( options )
		setPlacements( data );
		setError( null );
	} catch( err ) {
		setError( err );
	} finally {
		if ( ! quiet ) setInFlight( false );
	}
};
```

File: assets/wizards/advertising/views/placements/index.js [hunk 2]
```diff
@@ -65,44 +175,77 @@ const Placements = ( { adUnits } ) => {
...
+	const isEnabled = placementKey => {
+		return placements[ placementKey ].data?.enabled;
+	};
+
+	// Fetch initial placements and bidders.
 	useEffect( () => {
-		fetchPlacements();
+		const init = async () => {
+			await placementsApiFetch( { path: '/newspack-ads/v1/placements' } );
+			await apiFetch( { path: '/newspack-ads/v1/bidders' } )
```
Comment (dkoo) [inline assets/wizards/advertising/views/placements/index.js:228]: Same question as above: `await`/`async` with `try`/`catch`, or just promises?

File: assets/wizards/advertising/views/placements/index.js [hunk 3]
```diff
@@ -47,16 +51,122 @@ const getAdUnitsForSelect = adUnits => {
...
+		return ( sizes || [] ).find(
+			size => size[ 0 ] === sizeToCheck[ 0 ] && size[ 1 ] === sizeToCheck[ 1 ]
+		);
+	} );
+};
+
+/**
+ * Placement Control Component.
+ */
+const PlacementControl = ( {
```
Comment (dkoo) [inline assets/wizards/advertising/views/placements/index.js:72]: Non-blocking suggestion: does this component warrant its own module? It seems to have its own state and logic.

**Review by dkoo — APPROVED** [review]
Looking good—love the new UI. Just a few non-blocking suggestions/questions below.

File: assets/wizards/advertising/views/placements/index.js (hunk 1 above)
Comment (miguelpeixe) [reply assets/wizards/advertising/views/placements/index.js]: Good call! Done in 084b0cbce5c1577a39f85b55352730f95d3f3b45

File: assets/wizards/advertising/views/placements/index.js (hunk 2 above)
Comment (miguelpeixe) [reply assets/wizards/advertising/views/placements/index.js]: Same answer: 084b0cbce5c1577a39f85b55352730f95d3f3b45 😅

File: assets/wizards/advertising/views/placements/index.js (hunk 3 above)
Comment (miguelpeixe) [reply assets/wizards/advertising/views/placements/index.js]: Absolutely! It distributes the code nicely. 084b0cbce5c1577a39f85b55352730f95d3f3b45

**Review by dkoo — APPROVED** [review]
Thanks for the changes!

**Pull request merged**